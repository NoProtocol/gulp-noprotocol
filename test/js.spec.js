"use strict";
var gulp = require("gulp");
var streamAssert = require("stream-assert-gulp");
var noprocotol = require("../");
var fixtureDir = __dirname + "/fixtures/";

describe("noprocotol.js", function() {
  it("generates js and a sourcemap file", function(done) {
    gulp
      .src(fixtureDir + "controller.js")
      .pipe(
        noprocotol.js({
          annotate: true
        })
      )
      .pipe(streamAssert.length(2))
      .pipe(
        streamAssert.first(function(file) {
          expect(file.relative).toBe("controller.js.map");
        })
      )
      .pipe(
        streamAssert.second(function(file) {
          expect(file.relative).toBe("controller.js");
          expect(file.contents.toString()).toMatchSnapshot();
        })
      )
      .on("end", done);
  });
});
