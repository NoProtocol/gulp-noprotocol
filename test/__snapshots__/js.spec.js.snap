// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`noprocotol.js generates js and a sourcemap file 1`] = `
"\\"use strict\\";app.controller(\\"TestCtrl\\",[\\"$scope\\",function(t){t.x;return!0}]);
//# sourceMappingURL=controller.js.map
"
`;
